apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1308 @@
+void ws_test_chunk_transfer(struct http_client_request_t *param, 
+							int send_len, int sockfd, 
+							struct http_client_tls_t *client_tls) {
+	FUNC_EN;
+	int ret, cbuf, chunks = g_chunk_count - 1;
+	//send headers
+	int buf_len = 0;
+	int sndlen = send_len-g_last_clen;
+	while (sndlen > 0) {
+		if (param->tls) {
+			ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+									(const unsigned char *)param->buffer + buf_len,
+									sndlen);
+		} else {
+			ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+		}
+		if (ret < 1) {
+			PRNT("ERROR: send failed: %d", ret);
+			goto errout;
+		} else {
+			sndlen -= ret;
+			buf_len += ret;
+			PRNT("SEND SUCCESS: send %d bytes", ret);
+		}
+	}
+	//send chunks
+	cbuf = buf_len;
+	while(chunks--) {
+		buf_len = cbuf;
+		sndlen = g_normal_clen;
+		while (sndlen > 0) {
+			if (param->tls) {
+				ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+										(const unsigned char *)param->buffer + buf_len,
+										sndlen);
+			} else {
+				ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+			}
+			if (ret < 1) {
+				PRNT("ERROR: send failed: %d", ret);
+				goto errout;
+			} else {
+				sndlen -= ret;
+				buf_len += ret;
+				PRNT("SEND SUCCESS: send %d bytes", ret);
+			}
+		}
+	}
+	//last chunk
+	buf_len = cbuf;
+	sndlen = g_last_clen;
+	while (sndlen > 0) {
+		if (param->tls) {
+			ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+									(const unsigned char *)param->buffer + buf_len,
+									sndlen);
+		} else {
+			ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+		}
+		if (ret < 1) {
+			PRNT("ERROR: send failed: %d", ret);
+			goto errout;
+		} else {
+			sndlen -= ret;
+			buf_len += ret;
+			PRNT("SEND SUCCESS: send %d bytes", ret);
+		}
+	}
+
+errout:
+	if (param->callback && param->response) {
+		ws_test_http_client_response_release(param->response);
+	}
+	if (param->tls) {
+		ws_test_wget_tls_ssl_release(client_tls);
+	}
+	FUNC_EX;
+}
+
+static int ws_test_wget_base(void *arg)
+{
+	FUNC_EN;
+	int sockfd = -1;
+	int ret;
+	int buf_len, sndlen, len;
+	int encoding = CONTENT_LENGTH;
+	int state = HTTP_REQUEST_HEADER;
+	struct http_message_len_t mlen = {0,};
+	struct ws_test_wget_s ws;
+	struct http_client_request_t *param = (struct http_client_request_t *)arg;
+	struct http_client_response_t response = {0, };
+	int read_finish = false;
+	int send_len;
+
+	struct http_client_tls_t *client_tls = (struct http_client_tls_t *)malloc(sizeof(
+			struct http_client_tls_t));
+	int handshake_retry = WEBCLIENT_CONF_HANDSHAKE_RETRY;
+
+	if (client_tls == NULL) {
+		PRNT("memory not allocated");
+		return -1;
+	}
+
+	/* Initialize the state structure */
+	memset(&ws, 0, sizeof(struct ws_test_wget_s));
+	ws.buffer = param->buffer;
+	ws.buflen = param->buflen;
+
+	/* Parse the hostname (with optional port number) and filename from the URL */
+	ret = netlib_parsehttpurl(param->url, &ws.port, ws.hostname, WS_TEST_MAXHOSTNAME, ws.filename, WS_TEST_MAXFILENAME);
```

Comment:
better to add NETUTILS_NETLIB as dependency in KConfig. 
Or 
can this function call and dependency be removed
please check